src/main/java/noppes/npcs/scripted/ScriptWorld.java

```diff
@@ -119,6 +131,61 @@ public void removeBlock(int x, int y, int z){
 		world.setBlock(x, y, z, Blocks.air);
 	}
 	
+	public Vec3 rayCastBlockPos(Vec3 pos, Vec3 look, int maxDistance) {
+		Vec3 currentPos = pos; int rep = 0;
+		while (rep++ < maxDistance + 10) {
+			currentPos = currentPos.addVector(look.xCoord, look.yCoord, look.zCoord);
+			IBlock block = getBlock((int)currentPos.xCoord, (int)currentPos.yCoord, (int)currentPos.zCoord);
+			//System.out.println("Checking block at ["+(int)currentPos.xCoord+","+(int)currentPos.yCoord+","+(int)currentPos.zCoord+"]");
+			if (block == null) continue;
+			double distance = Math.pow(
+					Math.pow(currentPos.xCoord-pos.xCoord,2)
+					+Math.pow(currentPos.yCoord-pos.yCoord,2)
+					+Math.pow(currentPos.zCoord-pos.zCoord,2)
+					, 0.5);
+			//System.out.println("current distance check: "+distance+" on rep "+rep);
+			if (distance > maxDistance) return null;
+			return currentPos;
+		}
+		//System.out.println("ScriptWorld:WARNING: Repeated a ray cast to many times");
+		return null;
+	}
```

Comment:
Should return an IBlock object instead of a Vec3.

Reply:
I get that Vec3 isn't preferred because it would be annoying to use that in scripting. But IPos is integers and the look vector is supposed to be a normalized set of doubles and the position vector should be doubles as well. Should I make then double arrays instead?

Reply:
ended up making one with double[] parameters and another with IPos parameters